Source/santad/EventProviders/SNTEndpointSecurityManager.mm

```diff
@@ -240,6 +240,32 @@ - (void)messageHandler:(es_message_t *)m API_AVAILABLE(macos(10.15)) {
       callback = self.logCallback;
       break;
     }
+    case ES_EVENT_TYPE_AUTH_UNLINK: {
+      es_string_token_t pathToken = m->event.unlink.target->path;
+      NSString *path = [[NSString alloc] initWithBytes:pathToken.data
+                                                length:pathToken.length
+                                              encoding:NSUTF8StringEncoding];
+      if ([path isEqualToString:@"/private/var/db/santa/rules.db"] ||
+          [path isEqualToString:@"/private/var/db/santa/events.db"]) {
+        LOGW(@"Preventing attempt to delete Santa databases!");
+        es_respond_auth_result(self.client, m, ES_AUTH_RESULT_DENY, false);
```

Comment:
I think we can cache this, we are never going to allow unlinking or renaming for these paths.

Reply:
Good point